src/coreclr/jit/valuenum.cpp

```diff
@@ -8354,7 +8354,30 @@ void Compiler::fgValueNumberAssignment(GenTreeOp* tree)
         {
             // This means that there is an implicit cast on the rhs value
             // We will add a cast function to reflect the possible narrowing of the rhs value
-            rhsVNPair = vnStore->VNPairForCast(rhsVNPair, lhs->TypeGet(), rhs->TypeGet());
+            GenTree* effectiveLhs = lhs->gtEffectiveVal();
```

Comment:
`gtEffectiveVal` not needed, `ASG(COMMA(..., LCL_VAR), ...)` is not valid IR.

Reply:
Ok cool, glad we don't need to use that.